osquery/remote/http_client.cpp

```diff
@@ -29,6 +29,24 @@ const std::string kProxyDefaultPort{"3128"};
 
 const long kSSLShortReadError{0x140000dbL};
 
+void Client::callNetworkOperation(std::function<void()> callback) {
+  if (client_options_.timeout_) {
+    timer_.async_wait(
```

Comment:
Can this also use a `std::bind` instead of a lambda. I am not sure the pros/cons, but all of the other instances use `std::bind`.